### PR TITLE
fix: module federation plugin library should be optional

### PR DESCRIPTION
**Pull request by ScriptedAlchemy**
Addressing subtask on the Module Federation Project board

https://github.com/webpack/webpack/projects/6#card-33743655

**What kind of change does this PR introduce?**
Defaulting options from webpack core output config

fix/enhancement
**Did you add tests for your changes?**
none required?

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
ModuleFederationPlugin options and their default states

### CODE REVIEW

File: lib/container/ModuleFederationPlugin.js [hunk 1]
```diff
@@ -22,15 +22,17 @@ class ModuleFederationPlugin {
...
-			library: options.library,
+			library: options.library || options.library.type || output.library.type,
 			filename: options.filename,
 			exposes: options.exposes,
 			overridables: options.shared
 		}).apply(compiler);
 		new ContainerReferencePlugin({
-			remoteType: options.remoteType || options.library.type,
+			remoteType:
+				options.remoteType || options.library.type || output.library.type,
```
Comment (sokra) [inline lib/container/ModuleFederationPlugin.js:35]:
```suggestion
				options.remoteType || compiler.options.externalsType,
```

File: lib/container/ModuleFederationPlugin.js [hunk 2]
```diff
@@ -22,15 +22,17 @@ class ModuleFederationPlugin {
 	 */
 	apply(compiler) {
 		const { options } = this;
+		const { output } = compiler.options;
 		new ContainerPlugin({
 			name: options.name,
-			library: options.library,
+			library: options.library || options.library.type || output.library.type,
```
Comment (sokra) [inline lib/container/ModuleFederationPlugin.js:28]:
```suggestion
			library: options.library,
```

undefined will use the global defined library

File: lib/container/ModuleFederationPlugin.js [hunk 3]
```diff
@@ -30,7 +31,8 @@ class ModuleFederationPlugin {
 			overridables: options.shared
 		}).apply(compiler);
 		new ContainerReferencePlugin({
-			remoteType: options.remoteType || options.library.type,
+			remoteType:
+				options.remoteType || compiler.options.externalsType,
```
Comment (sokra) [inline lib/container/ModuleFederationPlugin.js:35]:
```suggestion
				options.remoteType || (options.library && options.library.type) || compiler.options.externalsType,
```

**Comment by ScriptedAlchemy** [comment]
@sokra Let me know if you've got any ideas what's going wrong here. My experience with systemjs is limited

**Comment by ScriptedAlchemy** [comment]
@evilebottnawi @sokra any ideas? Im blocked on this one

**Comment by sokra** [comment]
Thanks

**Pull request merged**